### PR TITLE
Feature/dx 480 make swish direct

### PR DESCRIPTION
**Pull request by marinafrydenb**

### CODE REVIEW

File: payments/swish/after-payment.md [hunk 1]
```diff
@@ -18,6 +20,36 @@ sidebar:
 
 {% include alert-development-section.md %}
 
+# **Swish Redirect and Payment Status**  
+
+After the payment is confirmed, the consumer will be redirected from the Swish
+app to the completeUrl set in the first API request `POST` [Create
```
Comment (asbjornu) [inline payments/swish/after-payment.md]:
```suggestion
app to the `completeUrl` set in the first API request `POST` [Create
```

File: payments/swish/after-payment.md [hunk 2]
```diff
@@ -18,6 +20,36 @@ sidebar:
...
+payment][create-payment] and you need to retrieve payment status with `GET`
+[Sales transaction][sales-transaction] before presenting a confirmation page to
+the consumer.
+
+## Options after posting a payment
+
+* **If CallbackURL is set**: Whenever changes to the payment occur a [Callback
+  request][technical-reference-callback] will be posted to the callbackUrl,
```
Comment (asbjornu) [inline payments/swish/after-payment.md]:
```suggestion
  request][technical-reference-callback] will be posted to the `callbackUrl`,
```

File: payments/swish/direct.md [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+        "payeeInfo": {
+            "payeeId": "bbb33dc5-f44e-4af6-afc0-27fb5fa2f63a",
+            "payeeReference": "ref-123456",
+            "payeeName": "Merchant1",
+            "productCategory": "A123",
+            "orderReference": "or-123456",
+            "subsite": "MySubsite"
+        },
+        "prefillInfo": {
+            "msisdn": "+46xxxxxxxxx"
```
Comment (asbjornu) [inline payments/swish/direct.md]: Use the full MSISDN from test data please

File: payments/swish/direct.md [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+
+## Sales
+
+The Sales resource lists the sales transactions (one or more) on a specific payment.
+
+{:.code-header}
+**Request**
+
+```http
+POST /psp/swish/payments HTTP/1.1
```
Comment (asbjornu) [inline payments/swish/direct.md]: Why is there a second `POST` request against `/psp/swish/payments` here without an HTTP body?

File: payments/swish/direct.md [hunk 5]
```diff
@@ -0,0 +1,335 @@
...
+Authorization: Bearer <AccessToken>
+Content-Type: application/json
+
+```
+
+{:.code-header}
+**Response**
+
+```http
+POST /psp/swish/payments HTTP/1.1
```
Comment (asbjornu) [inline payments/swish/direct.md]: These are a request, not response headers.

File: payments/swish/direct.md (hunk 4 above)
Comment (marinafrydenb) [reply payments/swish/direct.md]: corrected this to be a GET request

File: payments/swish/direct.md (hunk 5 above)
Comment (marinafrydenb) [reply payments/swish/direct.md]: I found them at this page https://developer.payex.com/xwiki/wiki/developer/view/Main/ecommerce/technical-reference/core-payment-resources/swish-payments/

File: payments/swish/direct.md [hunk 6]
```diff
@@ -0,0 +1,297 @@
...
+            "hostUrls": "https://example.com",
+            "paymentUrl": "http://example.com/perform-payment",
+            "completeUrl": "https://example.com/payment-completed",
+            "cancelUrl": "https://example.com/payment-canceled",
+            "callbackUrl": "https://example.com/payment-callback",
+            "logoUrl": "https://example.com/logo.png",
+            "termsOfServiceUrl": "https://example.com/terms.pdf"
+        },
+        "payeeInfo": {
+            "payeeId": "bbb33dc5-f44e-4af6-afc0-27fb5fa2f63a",
```
Comment (NullableInt) [inline payments/swish/direct.md]:
```suggestion
            "payeeId": "{{ page.merchantId }}",
```
Rebase on develop to have access to more variables.

File: payments/swish/direct.md [hunk 7]
```diff
@@ -0,0 +1,297 @@
...
+**Response**
+
+```http
+POST /psp/swish/payments HTTP/1.1
+Authorization: Bearer <AccessToken>
+Content-Type: application/json
+
+{
+    "payment": {
+        "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```
Comment (NullableInt) [inline payments/swish/direct.md]:
```suggestion
        "id": "/psp/swish/payments/{{ page.paymentId }}",
```
Replace all occurrences of this guid with the constant.

File: payments/swish/direct.md [hunk 8]
```diff
@@ -0,0 +1,297 @@
...
+            "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/urls"
+        },
+        "payeeInfo": {
+            "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/payeeinfo"
+        }
+    },
+    "operations": [
+        {
+            "method": "PATCH",
+            "href": "http://localhost:18496/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```
Comment (NullableInt) [inline payments/swish/direct.md]:
```suggestion
            "href": "http://{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```
Rebase on develop to have access to this variable.

File: payments/swish/direct.md [hunk 9]
```diff
@@ -0,0 +1,297 @@
...
+            "rel": "update-payment-abort"
+        },
+        {
+            "method": "POST",
+            "href": "http://localhost:18496/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```
Comment (NullableInt) [inline payments/swish/direct.md]:
```suggestion
            "href": "http://{{ page.apiUrl}}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```
Rebase on develop to have access to this variable.

File: payments/swish/direct.md [hunk 10]
```diff
@@ -0,0 +1,297 @@
...
+Content-Type: application/json
+
+{
+    "payment": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
+    "sale": {
+        "date": "23.10.2017 08:39:37 +00:00",
+        "paymentRequestToken": "LhXrK84MSpWU2RO09f8kUP-FHiBo-1pB",
+        "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales/6bf31479-623f-418a-d69e-08d519f19722",
+        "transaction": {
+            "id": "6bf31479-623f-418a-d69e-08d519f19722",
```
Comment (NullableInt) [inline payments/swish/direct.md]:
```suggestion
            "id": "{{ page.transactionId }}",
```

File: payments/swish/direct.md (hunk 7 above)
Comment (asbjornu) [reply payments/swish/direct.md]: This still needs to be done, it seems. :)

File: payments/swish/direct.md [hunk 11]
```diff
@@ -0,0 +1,297 @@
...
+            "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/urls"
+        },
+        "payeeInfo": {
+            "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/payeeinfo"
+        }
+    },
+    "operations": [
+        {
+            "method": "PATCH",
+            "href": "//{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```
Comment (asbjornu) [inline payments/swish/direct.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```

File: payments/swish/direct.md [hunk 12]
```diff
@@ -0,0 +1,297 @@
...
+            "rel": "update-payment-abort"
+        },
+        {
+            "method": "POST",
+            "href": "http://{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```
Comment (asbjornu) [inline payments/swish/direct.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```

File: payments/swish/direct.md (hunk 7 above)
Comment (marinafrydenb) [reply payments/swish/direct.md]: sorry, this is done now :)

Comment (asbjornu) [reply payments/swish/direct.md]: Have you forgotten to push? I can't see that it is done here on GitHub. :)

File: payments/swish/direct.md [hunk 13]
```diff
@@ -0,0 +1,297 @@
...
+    },
+    "operations": [
+        {
+            "method": "PATCH",
+            "href": "http://{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
+            "rel": "update-payment-abort"
+        },
+        {
+            "method": "POST",
+            "href": "http://{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```
Comment (asbjornu) [inline payments/swish/direct.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales",
```

File: payments/swish/direct.md [hunk 14]
```diff
@@ -0,0 +1,297 @@
...
+            "id": "/psp/swish/payments/{{ page.paymentId }}/urls"
+        },
+        "payeeInfo": {
+            "id": "/psp/swish/payments/{{ page.paymentId }}/payeeinfo"
+        }
+    },
+    "operations": [
+        {
+            "method": "PATCH",
+            "href": "http://{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```
Comment (asbjornu) [inline payments/swish/direct.md]:
```suggestion
            "href": "{{ page.apiUrl }}/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
```

File: payments/swish/direct.md [hunk 15]
```diff
@@ -0,0 +1,297 @@
...
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "payment": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
+    "sale": {
+        "date": "23.10.2017 08:39:37 +00:00",
+        "paymentRequestToken": "LhXrK84MSpWU2RO09f8kUP-FHiBo-1pB",
+        "id": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f/sales/6bf31479-623f-418a-d69e-08d519f19722",
```
Comment (asbjornu) [inline payments/swish/direct.md]: Please replace all GUIDs in URLs with `{{ page.paymentId }}` and similar constants. :)

```suggestion
        "id": "/psp/swish/payments/{{ page.paymentId }}",
```

File: payments/swish/direct.md [hunk 16]
```diff
@@ -0,0 +1,297 @@
...
+**Response**
+
+```http
+HTTP/1.1 200 OK
+Content-Type: application/json
+
+{
+    "payment": "/psp/swish/payments/20dfbcb9-587a-4ce9-e63e-08d519f1802f",
+    "sale": {
+        "date": "23.10.2017 08:39:37 +00:00",
```
Comment (asbjornu) [inline payments/swish/direct.md]: Is this `date` correct? The format looks very, very wrong. If this is the date format our Swish API outputs, it's a major bug.

File: payments/swish/direct.md (hunk 14 above)
Comment (asbjornu) [reply payments/swish/direct.md]: Can you please attend to this suggestion? :)

File: payments/swish/direct.md (hunk 13 above)
Comment (asbjornu) [reply payments/swish/direct.md]: Can you please attend to this suggestion? :)

File: payments/swish/direct.md (hunk 15 above)
Comment (asbjornu) [reply payments/swish/direct.md]: Can you please attend to this suggestion? :)

File: payments/swish/after-payment.md [hunk 17]
```diff
@@ -18,7 +20,37 @@ sidebar:
 
 {% include alert-development-section.md %}
 
-## Operations
+# **Swish Redirect and Payment Status**
```
Comment (asbjornu) [inline payments/swish/after-payment.md]:
```suggestion
# Swish Redirect and Payment Status
```

File: _includes/transaction-response.md [hunk 18]
```diff
@@ -1,7 +1,9 @@
 {% assign instrument = include.payment-instrument | default: "paymentorder" %}
 {% assign transaction = include.transaction | default: "capture" %}
 {% assign plural = transaction | append: "s" %}
+{% assign hiderequest = include.hiderequest | default: false %}
```
Comment (asbjornu) [inline _includes/transaction-response.md]:
```suggestion
{% assign showRequest = include.showRequest | default: true %}
```

File: _includes/transaction-response.md [hunk 19]
```diff
@@ -1,7 +1,9 @@
...
 
+{% unless hiderequest %}
```
Comment (asbjornu) [inline _includes/transaction-response.md]:
```suggestion
{% if showRequest %}
```

File: _includes/transaction-response.md [hunk 20]
```diff
@@ -12,6 +14,8 @@ Authorization: Bearer <AccessToken>
 Content-Type: application/json
 ```
 
+{% endunless %}
```
Comment (asbjornu) [inline _includes/transaction-response.md]:
```suggestion
{% endif %}
```

File: payments/swish/direct.md [hunk 21]
```diff
@@ -0,0 +1,268 @@
...
+
+{
+    "transaction": {
+        "msisdn": "+46739000001"
+    }
+}
+
+```
+
+{% include transaction-response.md hiderequest=true payment-instrument="swish" transaction="sale" %}
```
Comment (asbjornu) [inline payments/swish/direct.md]:
```suggestion
{% include transaction-response.md showRequest=false payment-instrument="swish" transaction="sale" %}
```

File: payments/swish/after-payment.md [hunk 22]
```diff
@@ -35,23 +67,23 @@ is given below.
...
         {
             "method": "POST",
-            "href": "https://{{ page.apiUrl }}/psp/swish/payments/3648fa94-7fd8-4e32-a14b-08d608f884ff/sales",
+            "href": "{{ page.apiUrl }}/psp/swish/payments/{{ page.paymentId }}/sales",
             "rel": "create-sale"
         },
         {
             "method": "GET",
-            "href": "https://{{ page.frontEndUrl }}/swish/payments/sales/993b479653da83671c074316c7455da05fced9d634431edbb64f3c5f80a863f0",
+            "href": "{{ page.frontEndUrl }}/swish/payments/sales/993b479653da83671c074316c7455da05fced9d634431edbb64f3c5f80a863f0",
```
Comment (asbjornu) [inline payments/swish/after-payment.md]:
```suggestion
            "href": "{{ page.frontEndUrl }}/swish/payments/sales/{{ page.paymentToken }}",
```

**Pull request merged**